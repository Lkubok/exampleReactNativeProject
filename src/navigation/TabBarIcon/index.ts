import Component from './TabBarIcon';

export default Component;
