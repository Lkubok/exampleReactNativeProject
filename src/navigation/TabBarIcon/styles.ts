import { StyleSheet } from 'react-native';
import { variables } from 'res';

export default StyleSheet.create({
  activeTick: {
    flex: 1,
    height: 2,
    position: 'absolute',
    top: 2,
    width: 15,
  },
  iconContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'flex-start',
    marginTop: variables.spacing.small,
    position: 'relative',
  },
  text: {
    fontSize: variables.fontSize.min,
    letterSpacing: -0.24,
  },
});
