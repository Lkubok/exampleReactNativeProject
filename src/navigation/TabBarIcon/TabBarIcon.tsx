import React, { FC, ReactNode, useMemo } from 'react';
import { View } from 'react-native';

import { screenNames, MainStackNames } from 'navigation';
import { variables } from 'res';

import MarketPlaceIcon from 'assets/tabBarMarketPlace.svg';
import NewsIcon from 'assets/tabBarNews.svg';
import InquiryIcon from 'assets/education.svg';
import ProfileIcon from 'assets/tabBarProfile.svg';

import styles from './styles';

interface Props {
  focused: boolean;
  routeName: MainStackNames;
  iconSize: number;
}

interface TabIconProps {
  children: ReactNode;
}

const TabBarIcon: FC<Props> = ({ routeName, iconSize, focused }) => {
  const iconColor = useMemo(() => (focused ? variables.colors.accentGreen : variables.colors.steel), [focused]);

  const TabIcon: FC<TabIconProps> = ({ children }) => {
    return <View style={styles.iconContainer}>{children}</View>;
  };

  switch (routeName) {
    case screenNames.MainStackNames.MarketPlaceStack:
      return (
        <TabIcon>
          <MarketPlaceIcon height={iconSize} width={iconSize} fill={iconColor} />
        </TabIcon>
      );
    case screenNames.MainStackNames.NewsStack:
      return (
        <TabIcon>
          <NewsIcon height={iconSize} width={iconSize} fill={iconColor} />
        </TabIcon>
      );
    case screenNames.MainStackNames.AdvisorStack:
      return (
        <TabIcon>
          <InquiryIcon height={iconSize} width={iconSize} fill={iconColor} />
        </TabIcon>
      );
    case screenNames.MainStackNames.ProfileStack:
      return (
        <TabIcon>
          <ProfileIcon height={iconSize} width={iconSize} fill={iconColor} />
        </TabIcon>
      );
    default:
      return (
        <TabIcon>
          <ProfileIcon height={iconSize} width={iconSize} color={iconColor} />
        </TabIcon>
      );
  }
};

export default TabBarIcon;
