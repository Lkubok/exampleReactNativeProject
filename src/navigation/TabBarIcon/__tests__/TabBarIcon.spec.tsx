import React from 'react';
import { act, customRender } from 'utils/testUtils';

import TabBarIcon from '../TabBarIcon';

jest.useFakeTimers();

describe('TabBarIcon component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      disabled: true,
      loading: false,
      onPress: () => {},
      title: 'testingTabBarIcon',
      useNativeDriver: true,
    };
  });

  it('should match snapshot', () => {
    const snapshot = customRender(<TabBarIcon {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
