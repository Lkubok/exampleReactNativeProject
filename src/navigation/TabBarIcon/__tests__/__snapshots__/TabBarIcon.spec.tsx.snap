// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon component should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginTop": 8,
        "position": "relative",
      }
    }
  >
    <SvgMock
      color="#8a8a8e"
    />
  </View>
</RNCSafeAreaProvider>
`;
